src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -3050,21 +3102,28 @@ private bool ReconsideredTypeAsAsyncModifier(ref SyntaxListBuilder modifiers, re
             return false;
         }
 
-        private bool TryParseIndexerOrPropertyDeclaration(SyntaxList<AttributeListSyntax> attributes, SyntaxListBuilder modifiers, TypeSyntax type,
-                                                          ExplicitInterfaceSpecifierSyntax explicitInterfaceOpt, SyntaxToken identifierOrThisOpt,
-                                                          TypeParameterListSyntax typeParameterListOpt, out MemberDeclarationSyntax result)
+        private bool TryParseIndexerOrPropertyDeclaration(
+            SyntaxList<AttributeListSyntax> attributes,
+            SyntaxListBuilder modifiers,
+            TypeSyntax type,
+            ExplicitInterfaceSpecifierSyntax explicitInterfaceOpt,
+            SyntaxToken identifierOrThisOpt,
+            TypeParameterListSyntax typeParameterListOpt,
+            out MemberDeclarationSyntax result)
         {
-            if (identifierOrThisOpt.Kind == SyntaxKind.ThisKeyword)
+            if (identifierOrThisOpt?.Kind == SyntaxKind.ThisKeyword)
             {
                 result = this.ParseIndexerDeclaration(attributes, modifiers, type, explicitInterfaceOpt, identifierOrThisOpt, typeParameterListOpt);
                 return true;
             }
 
             // `{` or `=>` definitely start a property.  Also allow
             // `; {` and `; =>` as error recovery for a misplaced semicolon.
```

Comment:
Consider removing this comment here since it was moved (with the logic) into IsStartOfPropertyBody

Reply:
351fbc4f135